pkg/processor/runtime/python/py/_nuclio_wrapper_as_server.py

```diff
@@ -0,0 +1,290 @@
+        await self._write_packet_to_processor(sock, 'r' + encoded_response)
+
+    async def _process_connection(self, sock):
+        """Process all events for a single connection."""
+        # signal start
+        self._logger.info("Signalling connection processing start")
+        await self._write_packet_to_processor(sock, 's')
+        self._logger.info(f"Event processing started for socket")
+        try:
+            while True:
+                # resolve event message length
+                event_message_length = await self._resolve_event_message_length(sock)
+                # Resolve event message
+                event = await self._resolve_event(sock, event_message_length)
+
+                # Handle the event
+                if not self._discard_events:
+                    try:
+                        await self._handle_event(event, sock)
+                    except BaseException as exc:
+                        await self._on_handle_event_error(exc, sock)
+                else:
+                    self._logger.debug("Event discarded")
+
+                # Release event reference
+                del event
+
+        except (ConnectionResetError, asyncio.IncompleteReadError):
+            self._logger.info("Client disconnected")
+        except WrapperFatalException as exc:
+            await self._on_serving_error(exc, sock)
+
+            # explode, unrecoverable exception
+            await self._shutdown(error_code=1)
+
+        except UnicodeDecodeError as exc:
+
+            # reset unpacker to avoid consecutive errors
+            # this may happen when msgpack fails to decode a non-utf8 events
+            self._unpacker = self._resolve_unpacker()
+            await self._on_serving_error(exc, sock)
+
+        except asyncio.CancelledError:
+            self._logger.debug('Waiting for event message was interrupted by a signal')
+        except Exception as exc:
+            await self._on_serving_error(exc, sock)
+        finally:
+            if self._is_drain_needed:
+                result = self._call_drain_handler()
+                if asyncio.iscoroutine(result):
+                    await result
+
+            if self._is_termination_needed:
+                result = self._call_termination_handler()
+                if asyncio.iscoroutine(result):
+                    await result
+            self.cleanup_connection(sock)
+
+    def cleanup_connection(self, sock):
+        """Cleanup resources for a disconnected client."""
+        fileno = sock.fileno()
+        if fileno in self.connections:
+            del self.connections[fileno]
+            self._logger.info(f"Connection {fileno} removed from tracking")
+        sock.close()
+
+        # Cancel and remove the associated task
+        if fileno in self.tasks:
+            task = self.tasks.pop(fileno)
+            if not task.done():
+                task.cancel()
+            self._logger.info(f"Task for connection {fileno} cancelled and removed")
```

Comment:
so the task is cancelling itself?

Reply:
@TomerShor reworked so when task is cancelling itself it will only close a socket without cancelling itself